src/nupic/encoders/CoordinateEncoder.cpp

```diff
@@ -0,0 +1,128 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2019, David McDougall
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implement the CoordinateEncoder
+ */
+
+#define _USE_MATH_DEFINES // Define before including anything
+#include <nupic/encoders/CoordinateEncoder.hpp>
+#include <math.h> // tgamma, M_PI
```

Comment:
nit,` <cmath> `

Reply:
Because errors on MSVC?

Reply:
because this is prefered way for c++; they're same same, but different. the <c*> work better with namespaces etc